### PR TITLE
fix(ivy): Run ChangeDetection on moved views

### PR DESCRIPTION
**Pull request by mhevery**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by AndrewKushnir** [comment]
Initial set of g3 presubmits to see how this PR affects g3 targets:
- [VE Presubmit](http://test/OCL:279016656:BASE:279016697:1573110401251:ae1c5b99)
- [Ivy Presubmit](http://test/OCL:279016656:BASE:279016806:1573110467915:1cad5204)
- [Global Ivy Presubmit](https://test.corp.google.com/OCL:279016656:BASE:279016960:1573110567958:11743276)

File: packages/core/src/render3/instructions/shared.ts [hunk 1]
```diff
@@ -1478,6 +1478,15 @@ function refreshDynamicEmbeddedViews(lView: LView) {
         ngDevMode && assertDefined(embeddedTView, 'TView must be allocated');
         refreshView(embeddedLView, embeddedTView, embeddedTView.template, embeddedLView[CONTEXT] !);
       }
+      const movedViews = viewOrContainer[MOVED_VIEWS];
```
Comment (pkozlowski-opensource) [inline packages/core/src/render3/instructions/shared.ts]: Cool, I can see that you could re-use `MOVED_VIEWS` that I've introduced for queries

File: packages/core/test/acceptance/view_container_ref_spec.ts [hunk 2]
```diff
@@ -1561,7 +1563,8 @@ describe('ViewContainerRef', () => {
       fixture.detectChanges();
       expect(log).toEqual([
         'doCheck-A', 'doCheck-B', 'doCheck-C', 'afterContentChecked-C', 'afterViewChecked-C',
-        'afterContentChecked-A', 'afterContentChecked-B', 'afterViewChecked-A', 'afterViewChecked-B'
+        'doCheck-C', 'afterContentChecked-C', 'afterViewChecked-C', 'afterContentChecked-A',
+        'afterContentChecked-B', 'afterViewChecked-A', 'afterViewChecked-B'
```
Comment (pkozlowski-opensource) [inline packages/core/test/acceptance/view_container_ref_spec.ts]: hmm, are those tests still passing  in both ivy and VE after the change? If so, why it is the case (bad tests?)

Comment (mhevery) [reply packages/core/test/acceptance/view_container_ref_spec.ts]: fixed and reverted.

File: packages/core/src/render3/instructions/shared.ts (hunk 1 above)
Comment (mhevery) [reply packages/core/src/render3/instructions/shared.ts]: Yes but it turns out it is not exactly what we need.

**Comment by mhevery** [comment]
[presubmit](http://test/OCL:279326390:BASE:279326428:1573233010393:a5c671a7)
[ivy presubmit](http://test/OCL:279326390:BASE:279326612:1573233084961:dabd247e)

**Comment by mhevery** [comment]
[presubmit](http://test/OCL:279327300:BASE:279327328:1573233338582:2950874a)

File: packages/core/src/render3/instructions/shared.ts [hunk 3]
```diff
@@ -1481,6 +1481,32 @@ function refreshDynamicEmbeddedViews(lView: LView) {
...
+        // the component where they were declared. Moved views also contains intra component moves,
+        // which we don't care about.
+        // TODO(misko): this is not the most efficient way to do this as we have to do a lot of
+        // searches. Will refactor for performance later.
+        const declaredComponentLView = findComponentView(lView);
+        for (let i = 0; i < movedViews.length; i++) {
+          const movedLView = movedViews[i] !;
+          let parentLView = movedLView[PARENT];
+          while (isLContainer(parentLView)) {
+            parentLView = parentLView[PARENT];
```
Comment (pkozlowski-opensource) [inline packages/core/src/render3/instructions/shared.ts]: Why do we need this logic of going up the parent nodes? Wouldn't `findComponentView(movedLView)` take care of it?

**Comment by pkozlowski-opensource** [comment]
LGTM

**Review by kara — APPROVED** [review]
LGTM, one nit

File: packages/core/src/render3/instructions/shared.ts [hunk 4]
```diff
@@ -1481,6 +1481,32 @@ function refreshDynamicEmbeddedViews(lView: LView) {
         ngDevMode && assertDefined(embeddedTView, 'TView must be allocated');
         refreshView(embeddedLView, embeddedTView, embeddedTView.template, embeddedLView[CONTEXT] !);
       }
+      const movedViews = viewOrContainer[MOVED_VIEWS];
+      if (movedViews !== null) {
```
[the remark this answers is not in the document]
Comment (mhevery) [reply packages/core/src/render3/instructions/shared.ts]: Good idea. Is it OK that I have done it in the follow up PR. https://github.com/angular/angular/pull/33702/commits/e46d7a2fc730c5f7f5bee49afe64a5e324599b08 The https://github.com/angular/angular/pull/33702 does whole bunch of cleanup already in that function.

[the remark this answers is not in the document]
Comment (kara) [reply packages/core/src/render3/instructions/shared.ts]: Sounds good to me!

**Pull request closed without merging**